File: pythonlibrary/aibrd/generators/test_cases.py
from datetime import date
from ..models.outputs import BRDContent


def generate_test_cases(content: BRDContent) -> str:
    today = date.today().isoformat()
    prefix = content.module_prefix or ""
    lines = ["# Test Cases", f"_Generated: {today}_", ""]

    for ac in content.criteria:
        tc_id = ac.id.replace("AC", f"{prefix}-TC" if prefix else "TC")
        lines += [
            f"## {tc_id}",
            f"_Traces: {ac.id}{', ' + ac.related_flow if ac.related_flow else ''}_",
            "",
            "```gherkin",
            f"Scenario: {ac.when}",
            f"  Given {ac.given}",
            f"  When {ac.when}",
            f"  Then {ac.then}",
            "```",
            "",
        ]

    for rule in content.rules:
        tc_id = rule.id.replace("BR", f"{prefix}-TC" if prefix else "TC")
        lines += [
            f"## {tc_id}-boundary",
            f"_Traces: {rule.id}_",
            "",
            "```gherkin",
            f"Scenario: Verify rule — {rule.description[:60]}",
            "  Given the system is operational",
            "  When the condition applies",
            f"  Then the rule is enforced: {rule.description}",
            "```",
            "",
        ]

    return "\n".join(lines)

File: vscodebase/src/core/generators/test_cases.ts
import { BRDContent } from '../models/outputs'

export function generateTestCases(content: BRDContent, modulePrefix?: string): string {
  const lines: string[] = [
    '# Test Cases',
    `_Generated: ${new Date().toISOString().split('T')[0]}_`,
    ''
  ]

  for (const ac of content.criteria) {
    const tcId = ac.id.replace(/^AC/, modulePrefix ? `${modulePrefix}-TC` : 'TC')
    lines.push(`## ${tcId}`)
    lines.push(`_Traces: ${ac.id}${ac.relatedFlow ? `, ${ac.relatedFlow}` : ''}_`, '')
    lines.push('```gherkin')
    lines.push(`Scenario: ${ac.when}`)
    lines.push(`  Given ${ac.given}`)
    lines.push(`  When ${ac.when}`)
    lines.push(`  Then ${ac.then}`)
    lines.push('```', '')
  }

  for (const rule of content.rules) {
    const tcId = rule.id.replace(/^BR/, modulePrefix ? `${modulePrefix}-TC` : 'TC')
    lines.push(`## ${tcId}-boundary`)
    lines.push(`_Traces: ${rule.id}_`, '')
    lines.push('```gherkin')
    lines.push(`Scenario: Verify rule — ${rule.description.slice(0, 60)}`)
    lines.push(`  Given the system is operational`)
    lines.push(`  When the condition applies`)
    lines.push(`  Then the rule is enforced: ${rule.description}`)
    lines.push('```', '')
  }

  return lines.join('\n')
}
